akka-stream/src/main/scala/akka/stream/stage/Stage.scala

```diff
@@ -531,6 +532,29 @@ sealed trait TerminationDirective extends SyncDirective
 // never instantiated
 sealed abstract class FreeDirective private () extends UpstreamDirective with DownstreamDirective with TerminationDirective with AsyncDirective
 
+object LifecycleContext {
+  private final val DefaultLoggerName = "akka.stream.Log"
+
+  /**
+   * INTERNAL API
+   * Must be located here to be visible for implicit resolution, when LifecycleContext is passed to [[Logging]]
+   * More specific LogSource than `fromString`, which would add the ActorSystem name in addition to the supervision to the log source.
+   */
+  implicit private[akka] final val fromLifecycleContext = new LogSource[LifecycleContext] {
```

Comment:
Ah, that can be passed in explicitly actually - good idea (`Logging(sys, ctx)(fromLivecycleContext)`) 
I somehow missed that it can be explicitly passed in